### PR TITLE
Flushed server handling.

### PR DESCRIPTION
**Pull request by n1k0**
Supersedes #209.

- [x] `Collection#sync()` rejection on flushed server detected
- [x] `Collection.resetSyncStatus()` implementation
- [x] Integration tests for both
- [x] Documentation
- [x] Update dist files

### CODE REVIEW

**Comment by n1k0** [comment]
This is now ready for final review. r=? @leplatrem

Also requesting feedback from @michielbdejong.

File: docs/api.md [hunk 1]
```diff
@@ -391,6 +391,30 @@ function sync() {
...
+
+So instead of wiping your local database to reflect this new remote state, you get a proper error :) In this situation, you'll probably want to republish your local database to the server; this is very easy to achieve by calling [`#resetSyncStatus()`](https://doc.esdoc.org/github.com/Kinto/kinto.js/class/src/collection.js~Collection.html#instance-method-resetSyncStatus), then [`#sync()`](https://doc.esdoc.org/github.com/Kinto/kinto.js/class/src/collection.js~Collection.html#instance-method-sync) again:
+
+```js
+articles.sync()
+  .catch(err => {
+    if (err.message.contains("flushed")) {
+      return articles.resetSyncStatus()
+        .then(_ => articles.sync());
+    } else {
```
Comment (leplatrem) [inline docs/api.md]: nit: superflous else

File: docs/api.md [hunk 2]
```diff
@@ -391,6 +391,30 @@ function sync() {
...
+      throw err;
+    }
+  })
+  .then(result => {
+    console.log(result);
+  })
+  .catch(error => {
+    console.error(error);
+  });
```
Comment (leplatrem) [inline docs/api.md]: nit: in other parts of the docs, `console.error.bind(console)` is used  (I personnaly prefer this form, but would recommand consistency).

File: test/integration_test.js [hunk 3]
```diff
@@ -23,26 +23,30 @@ describe("Integration tests", () => {
...
+        serverLogs.push(data);
+      });
+      server.on("close", code => {
+        if (code && code > 0) {
+          new Error("Server errors encountered:\n" +
+            serverLogs.map(line => line.toString()).join(""));
+        }
+      });
+      // Allow some time for the server to start.
+      setTimeout(resolve, 1000);
```
Comment (leplatrem) [inline test/integration_test.js]: :miam: ;)

File: test/integration_test.js [hunk 4]
```diff
@@ -1080,13 +1088,52 @@ describe("Integration tests", () => {
...
+        .should.be.rejectedWith(Error, "Server has been flushed");
     });
 
-    after(() => {
-      return stopServer();
+    it("should allow republishing local collection to flushed server", () => {
+      return tasks.sync()
+        .catch(_ => tasks.resetSyncStatus())
+        .then(_ => tasks.sync())
+        .should.eventually.have.property("published").to.have.length.of(2);
```
Comment (leplatrem) [inline test/integration_test.js]: :+1:

**Comment by leplatrem** [comment]
r+ ! 

> As a side discussion (no very strong use-case so far), changing remote server on an existing non-empty bucket/collection will require smarter code to be detected. One possibility could be to store the remote URI in the metadata along the last-modified, and detect server change like that... But that's a different story of course...

**Pull request merged**